From a433d177585c8355c6ede67469a521d05cf1e9d7 Mon Sep 17 00:00:00 2001
From: Colin Watson <cjwatson@ubuntu.com>
Date: Fri, 17 Jan 2014 02:30:52 +0000
Subject: [PATCH 017/143] freebsd/hostdisk.c is only ever compiled on FreeBSD

* grub-core/osdep/freebsd/hostdisk.c (grub_util_fd_open): Remove
redundant preprocessor conditional.
---
 ChangeLog                          | 5 +++++
 grub-core/osdep/freebsd/hostdisk.c | 2 --
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/ChangeLog b/ChangeLog
index b445f12..6613b4e 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,8 @@
+2014-01-17  Colin Watson  <cjwatson@debian.org>
+
+	* grub-core/osdep/freebsd/hostdisk.c (grub_util_fd_open): Remove
+	redundant preprocessor conditional.
+
 2014-01-08  Colin Watson  <cjwatson@ubuntu.com>
 
 	* Makefile.util.def (grub-macbless): Change mansection to 8.
diff --git a/grub-core/osdep/freebsd/hostdisk.c b/grub-core/osdep/freebsd/hostdisk.c
index bd5fddb..eb202dc 100644
--- a/grub-core/osdep/freebsd/hostdisk.c
+++ b/grub-core/osdep/freebsd/hostdisk.c
@@ -108,7 +108,6 @@ grub_util_fd_open (const char *os_dev, int flags)
 
   ret = open (os_dev, flags, S_IROTH | S_IRGRP | S_IRUSR | S_IWUSR);
 
-#if defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
   if (! (sysctl_oldflags & 0x10)
       && sysctlbyname ("kern.geom.debugflags", NULL , 0, &sysctl_oldflags, sysctl_size))
     {
@@ -116,7 +115,6 @@ grub_util_fd_open (const char *os_dev, int flags)
       close (ret);
       return GRUB_UTIL_FD_INVALID;
     }
-#endif
 
   return ret;
 }
-- 
1.9.3

